packages/next/src/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -227,7 +227,7 @@ function getCreateAssets(params: {
           name,
           filePath,
         })),
-        environments: opts.edgeEnvironments,
+        env: opts.edgeEnvironments,
```

Comment:
Same as https://github.com/vercel/vercel/pull/11549/files#r1591393572

Reply:
It was not working properly with #64108, it requires to work with some of the changes from the PR. Consider it was only landed on canary and only changed preview props, we want to do the renaming and fix together